webui/react/src/components/MetricSelectFilter.tsx

```diff
@@ -37,9 +39,31 @@ const MetricSelectFilter: React.FC<Props> = ({ metricNames, multiple, onChange,
     return metricNames.filter(metric => metric.type === MetricType.Validation);
   }, [ metricNames ]);
 
+  const totalNumMetrics = useMemo(() => {
+    return metricNames.length;
+  }, [ metricNames ]);
+
+  const filterFn = (search: string, metricName: string) => {
+    return metricName.toLocaleLowerCase().indexOf(search.toLocaleLowerCase()) !== -1;
+  };
```

Comment:
**chore:** Since this function has no dependencies on anything inside the `MetricSelectFilter`, it can be moved just above the `const MetricSelectFilter: React...` line. This way it won't get recreated each time `MetricSelectFilter` rerenders.